src/useFormState.ts

```diff
@@ -113,9 +115,11 @@ export function useFormState<T, I>(opts: UseFormStateOpts<T, I>): ObjectState<T>
               // user's autoSave function itself wants to call a .set.
               const promise = autoSaveRef.current!(form);
               isAutoSaving = "in-flight";
+              autoSaveStatusContext.triggerAutoSave();
               await promise;
             } finally {
               isAutoSaving = false;
+              autoSaveStatusContext.resolveAutoSave();
```

Comment:
Wondering if we need to/should teach FormState about errors here too. Then a user could set up their form state's auto save to throw the error/reject the promise and have a `catch` method here that'd take the error. At that point we might also need to pause any other pending saves to make sure any saves we have in flight aren't reliant on the order that they are executed in.

Reply:
Yeah that's worth a callout. I have a lot of infrastructure for displaying errors but so far nothing is actually passing errors in. Since maybeAutoSave codified try/finally (no catch) I'm not sure where to notify AutoSaveStatus that an error happened.

Reply:
I wouldn't say it's codified. It can always be updated and enhanced, we just want to try and avoid breaking existing use cases. And if it needs to be a breaking change, so be it. I'm hoping we can create some auto save utility in Blueprint to help glue our numerous mutations in with FormState and error handling.